src/validators/ArrayValidator.ts

```diff
@@ -95,6 +106,17 @@ export class ArrayValidator<T> extends BaseValidator<T[]> {
 			? Result.ok(transformed)
 			: Result.err(new CombinedPropertyError(errors));
 	}
+
+	private isUnique(givenValues: unknown[]) {
+		if (givenValues.length < 2) return true;
+
+		for (let index = 1; index <= givenValues.length; index++) {
+			const previousValue = givenValues[index - 1];
+			if (isDeepStrictEqual(previousValue, givenValues[index])) return false;
+		}
+
+		return true;
+	}
```

Comment:
@vladfrangu  @kyranet can you guys sanity check this change I made? Tests pass as expected.

Reply:
previous clone and shift were unnecessary. Idk why I had cloned it 😄

Reply:
This won't work if you do `[1, 2, 3, 2]` will it?